File: JS/np speed test.js
const time = Date.now()
let np = [2]

const npatrouver = 1000000

let nombre = 3

while (np.length < npatrouver) {
    let test = 0

    let max = Math.sqrt(nombre)

    while (np[test] <= max) {
        if (nombre % np[test] == 0) {
            nombre += 2
            max = Math.sqrt(nombre)
            test = 0
        } else {
            test++
        }
    }

    np.push(nombre)
    nombre += 2
}
console.log(nombre - 2)
console.log(npatrouver, "nombres premiers trouvés", Date.now() - time, "ms")

File: PY/np speed test.py
from time import time
from math import sqrt

time2 = time()
np = [2]

npatrouver = 100000
nplen = 1

nombre = 3

while (nplen < npatrouver):
    test = 0

    max = sqrt(nombre)
    while (np[test] <= max):
        if (nombre % np[test] == 0):
            nombre += 2
            max = sqrt(nombre)
            test = 0
        else:
            test += 1

    np.append(nombre)
    nplen += 1
    nombre += 2

print(npatrouver, "nombres premiers trouvés", time() - time2, "ms")
